src/github/issueOverview.ts

```diff
@@ -0,0 +1,416 @@
+		// Otherwise, create a new panel.
+		if (IssueOverviewPanel.currentPanel) {
+			IssueOverviewPanel.currentPanel._panel.reveal(activeColumn, true);
+		} else {
+			const title = `Issue #${issue.githubNumber.toString()}`;
+			IssueOverviewPanel.currentPanel = new IssueOverviewPanel(extensionPath, activeColumn || vscode.ViewColumn.Active, title, pullRequestManager, descriptionNode);
+		}
+
+		await IssueOverviewPanel.currentPanel!.update(issue, descriptionNode);
+	}
+
+	public static refresh(): void {
+		if (this.currentPanel) {
+			this.currentPanel.refreshPanel();
+		}
+	}
+
+	protected constructor(extensionPath: string, column: vscode.ViewColumn, title: string, pullRequestManager: PullRequestManager, descriptionNode: DescriptionNode, type: string = IssueOverviewPanel._viewType) {
+		this._extensionPath = extensionPath;
+		this._pullRequestManager = pullRequestManager;
+		this._descriptionNode = descriptionNode;
+
+		// Create and show a new webview panel
+		this._panel = vscode.window.createWebviewPanel(type, title, column, {
+			// Enable javascript in the webview
+			enableScripts: true,
+			retainContextWhenHidden: true,
+
+			// And restrict the webview to only loading content from our extension's `media` directory.
+			localResourceRoots: [
+				vscode.Uri.file(path.join(this._extensionPath, 'media'))
+			]
+		});
+
+		// Listen for when the panel is disposed
+		// This happens when the user closes the panel or when the panel is closed programatically
+		this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+
+		// Handle messages from the webview
+		this._panel.webview.onDidReceiveMessage(async message => {
+			await this._onDidReceiveMessage(message);
+		}, null, this._disposables);
+
+		this._pullRequestManager.onDidChangeActiveItem(_ => {
+			if (this._pullRequestManager && this._item) {
+				const isCurrentlyCheckedOut = this._item.equals(this._pullRequestManager.activeItem);
+				this._postMessage({
+					command: 'pr.update-checkout-status',
+					isCurrentlyCheckedOut: isCurrentlyCheckedOut
+				});
+			}
+		}, null, this._disposables);
+
+		onDidUpdatePR(pr => {
+			if (pr) {
+				this._item.update(pr);
+			}
+
+			this._postMessage({
+				command: 'update-state',
+				state: this._item.state,
+			});
+		}, null, this._disposables);
+	}
+
+	public async refreshPanel(): Promise<void> {
+		if (this._panel && this._panel.visible) {
+			this.update(this._item, this._descriptionNode);
+		}
+	}
+
+	public async updateIssue(issueModel: IssueModel, descriptionNode: DescriptionNode): Promise<void> {
+		return Promise.all([
+			this._pullRequestManager.resolveIssue(
+				issueModel.remote.owner,
+				issueModel.remote.repositoryName,
+				issueModel.githubNumber
+			),
+			this._pullRequestManager.getIssueTimelineEvents(issueModel),
+			this._pullRequestManager.getPullRequestRepositoryDefaultBranch(issueModel),
+		]).then(result => {
+			const [pullRequest, timelineEvents, defaultBranch] = result;
```

Comment:
local variables and error messages should use "issue" here

Reply:
Changed